internal/server/iigointernal/legislature.go

```diff
@@ -208,29 +222,47 @@ func (l *legislature) updateRules(ruleMatrix rules.RuleMatrix, ruleIsVotedIn boo
 // appointNextJudge returns the island ID of the island appointed to be Judge in the next turn
 func (l *legislature) appointNextJudge(monitoring shared.MonitorResult, currentJudge shared.ClientID, allIslands []shared.ClientID) (shared.ClientID, error) {
 	var election voting.Election
-	var nextJudge shared.ClientID
-	electionsettings := l.clientSpeaker.CallJudgeElection(monitoring, l.judgeTurnsInPower, allIslands)
-	if electionsettings.HoldElection {
+	var appointedJudge shared.ClientID
+	electionSettings := l.clientSpeaker.CallJudgeElection(monitoring, int(l.gameState.IIGOTurnsInPower[shared.Judge]), allIslands)
```

Comment:
Why cast to an int if it is stored as uint?

Reply:
Did not want to change agent function parameters, but will do it now

Reply:
I can change the function signature, the value should not even be passed there. gameState.IIGOTurnsInPower is accessible through the clientGameState. Will do this in a separate PR

Reply:
Ahh, as it's client side, I'd prefer for you not to change this.

Reply:
just saw what lh said on discord!